bench/benchmark.hpp

```diff
@@ -0,0 +1,835 @@
+                                           fmt::format("({}, {}, {:4.1f} %)", to_si_prefix(branch_misses, "", 0),
+                                                       to_si_prefix(branch_total, "", 0), branch_ratio));
+                    result_map.try_emplace("CPU-I",
+                                           fmt::format("{}", to_si_prefix(instructions / n_iterations, "", 0)));
+                }
+
+                result_map.try_emplace("total time: ", to_si_prefix(duration_s));
+                result_map.try_emplace("ops/s", to_si_prefix(1.0 / duration_s, "", 2));
+
+                if constexpr (MARKER_SIZE > 0) {
+                    auto transposed_map = utils::convert(marker_iter);
+                    for (int keyID = 0; keyID < transposed_map.size(); keyID++) {
+                        if (keyID > 0) {
+                            const auto meas = fmt::format("  └─Marker{}: '{}'->'{}' ", keyID, transposed_map[0].first,
+                                                          transposed_map[keyID].first);
+                            auto &marker_result_map = ResultType::add_result(meas);
+
+                            marker_result_map.try_emplace("n-iter", "");
+                            add_statistics(marker_result_map, utils::diff<long double>(transposed_map[keyID].second,
+                                                                                       transposed_map[0].second));
+                            // add zero info to not duplicate the results from the superordinate benchmark
+                            if (PerformanceCounter::available()) {
+                                result_map.try_emplace("CPU cache misses", "");
+                                result_map.try_emplace("CPU branch misses", "");
+                                result_map.try_emplace("CPU-I", "");
+                            }
+                            result_map.try_emplace("total time: ", "");
+                            result_map.try_emplace("ops/s", "");
+                        }
+                    }
+                }
+
+            };
+            return *this;
+        }
+
+        template<size_t n>
+        auto repeat() {
+            return ::benchmark::benchmark<n, ResultType, meas_marker_names...>(this->name);
+        }
+    };
+
+    [[nodiscard]] auto operator ""_benchmark(const char *name, std::size_t size) {
+        return ::benchmark::benchmark<1LU>{{name, size}};
+    }
+
+}  // namespace benchmark
+
+
+
+namespace cfg {
+    namespace ut = boost::ut;
+
+    template<class TPrinter = ut::printer>
+    class reporter {
+        struct {
+            std::size_t fail{};
+        } _benchmarks{};
+
+        struct {
+            std::size_t pass{};
+            std::size_t fail{};
+        } _asserts{};
+        TPrinter _printer{};
+
+    public:
+        constexpr reporter &operator=(TPrinter printer) {
+            _printer = static_cast<TPrinter &&>(printer);
```

Comment:
std::move?

Reply:
Noted, this was copied from the original UT source...  some may say that std::move is a glorified 'cast'.

Will also shift more of the printout mechanics from above to this class to better separate benchmark state/result from it's visualisation for future alternate representations. Just realised this when tinkering with some of the upstream UT code.